From 617f8fedd10501f895e60107aaf51bcf66c6e7b1 Mon Sep 17 00:00:00 2001
From: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Date: Wed, 13 Nov 2024 09:08:09 +0100
Subject: [PATCH 1381/1695] plat-stm32mp2: set copro encryption test flag to
 insecure dev

The flag CFG_REMOTEPROC_ENC_TESTKEY must be dedicated to insecure dev
configuration and is disabled by default.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f9a54e406cd72c7fccedf400bef4b855bd50e284 plat-stm32mp2: set copro encryption test flag to insecure dev]
Change-Id: I6b334b46a7b87de731d34ba1045fc9e7cd4eeec2
Signed-off-by: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/418316
Tested-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 00dc3bb15..9da924b0c 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -252,6 +252,13 @@ RPROC_SIGN_KEY ?= keys/default.pem
 # Increase the RESERVED VA SPACE to be able to map reserved-memory regions
 # assigned to the remote processor.
 CFG_RESERVED_VASPACE_SIZE = 32 * 1024 * 1024
+
+# Co-processor encryption using test key is dedicated to insecure development
+# configuration only.
+CFG_REMOTEPROC_ENC_TESTKEY ?= n
+ifeq ($(CFG_REMOTEPROC_ENC_TESTKEY),y)
+$(call force,CFG_INSECURE,y,Required by CFG_REMOTEPROC_ENC_TESTKEY)
+endif
 endif
 
 # Default enable HWRNG PTA support
-- 
2.47.2

